src/sentry/tasks/base.py

```diff
@@ -91,6 +92,10 @@ def instrumented_task(name, stat_suffix=None, silo_mode=None, record_timing=Fals
     - disabling of result collection.
     """
 
+    # Use option to control default behavior of queue time monitoring
+    # Value can be dynamically updated, which is why the evaluation happens during function run-time
+    record_timing = record_timing or options.get("sentry-metrics.monitor-queue-time")
```

Comment:
I think this actually needs to be inside of `_wrapped` to work correctly. Otherwise, this will be assessed just when the `instrumented_task` decorator is first called, which is on module load.

Reply:
good catch